### PR TITLE
add metadata support to pyformance wrapper

### PR DESCRIPTION
**Pull request by charliesignalfx**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -230,19 +230,26 @@ metric registry data directly to SignalFx.
 
 .. code:: python
 
-    from pyformance import count_calls, gauge
+    from signalfx.pyformance import count_calls, gauge, count_calls_with_dims
```
Comment (mpetazzoni) [inline README.rst]: Might as well import `SignalFxReporter` along with it so you don't need the separate `import` line below (and change line 246)

File: signalfx/pyformance/metadata.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+class MetricMetadata(object):
```
Comment (mpetazzoni) [inline signalfx/pyformance/metadata.py]: All you files need copyright statements at the top

File: signalfx/pyformance/metadata.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    def __init__(self):
+        self._metadata = {}
+
+    def get_metadata(self, key):
+        """returns metadata for the specified key"""
+        dimensions = self._metadata.get(key)
+        return dimensions or {}
+
+    def register(self, key, gauge=None, **kwargs):
```
Comment (mpetazzoni) [inline signalfx/pyformance/metadata.py]: You don't appear to be using `gauge` in there

File: signalfx/pyformance/metadata.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        """registers metadata for a metric and returns a composit key"""
```
Comment (mpetazzoni) [inline signalfx/pyformance/metadata.py]: *composite

File: signalfx/pyformance/metadata.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+    It then makes this information available to the SignalFxReporter so metrics
+    can be reported with the appropriate metric and dimensions.
+    """
+
+    def __init__(self):
+        self._metadata = {}
+
+    def get_metadata(self, key):
+        """returns metadata for the specified key"""
```
Comment (mpetazzoni) [inline signalfx/pyformance/metadata.py]: *Returns metadata

File: signalfx/pyformance/metadata.py (hunk 4 above)
Comment (mpetazzoni) [inline signalfx/pyformance/metadata.py]: *Registers metadata

File: signalfx/pyformance/registry.py [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+        """clears the registered metrics and metadata"""
+        self.metadata.clear()
+        super(MetricsRegistry, self).clear()
+
+
+# set global registry on import to the SignalFx MetricsRegistry
+set_global_registry(MetricsRegistry())
+
+
+class RegexRegistry(MetricsRegistry):
```
Comment (mpetazzoni) [inline signalfx/pyformance/registry.py]: What is this for? The documentation is copy/pasted from the other one, so I don't know how it's different

Comment (charliesignalfx) [reply signalfx/pyformance/registry.py]: I've updated the documentation to describe what it is used for.  Essentially it is used for pattern matching methods in REST APIs

File: signalfx/pyformance/metadata.py (hunk 2 above)
Comment (charliesignalfx) [reply signalfx/pyformance/metadata.py]: I've added copy right statements to all of the new files

**Pull request merged**